tools/bug-report/pkg/config/config.go

```diff
@@ -162,6 +162,10 @@ type BugReportConfig struct {
 	// If set, StartTime must be unset.
 	Since Duration `json:"since,omitempty"`
 
+	// TimeFilterApplied stores if user has provided any time filtering flags.
+	// If Since, StartTime, EndTime are all not applied by the user, set TimeFilterApplied as false; Otherwise set true
+	TimeFilterApplied bool `json:"timeFilterApplied,omitempty"`
```

Comment:
I believe these are configurations that are set by user-facing command flags. It will be better to assign them to a global variable or use a function instead of including them in the `BugReportConfig`.

Reply:
I thought the [flags.go](https://github.com/istio/istio/blob/fe62d5b332a18bf183803702b17487584193227d/tools/bug-report/pkg/bugreport/flags.go#L31) is for user facing flags while the config.go is more for internal use. Is it not?

Reply:
I think its reasonable here. Slight preference to make it method on BugReportConfig though (since it is derived from the rest)

Reply:
In flag.go they would alter the startTime and endTime value the user passes in, so it might hard to make it a method on BugReportConfig because BugReportConfig don't have the original startTime and endTime maintained there.